### PR TITLE
job-security: init at unstable-0-2024-03-24

### PR DESCRIPTION
**Pull request by fgaz**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/jo/job-security/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitHub {
+    owner = "yshui";
+    repo = "job-security";
+    rev = "50163cb4259b86aa3c9bd4eaccef83d73a0747db";
+    hash = "sha256-OXLL6jMCc+1XkOZjK/FQY1hBthi1CkWt/CX8uqf3C/E=";
+  };
+
+  cargoHash = "sha256-s/oL2NYZXPVQSzdH3p815156kYmLJ294edh7vBzRNT4=";
+
+  # unresolved import `ptyprocess`
```
Comment (SuperSandro2000) [inline pkgs/by-name/jo/job-security/package.nix:20]: Why not add the python package to nativeCheckInputs?

Comment (fgaz) [reply pkgs/by-name/jo/job-security/package.nix:20]: It doesn't appear to be related to the python package:

```
running 1 test
test server/src/pty.rs - pty::PtyProcess::spawn (line 48) - compile ... FAILED

failures:

---- server/src/pty.rs - pty::PtyProcess::spawn (line 48) stdout ----
error[E0432]: unresolved import `ptyprocess`
 --> server/src/pty.rs:50:5
  |
4 | use ptyprocess::PtyProcess;
  |     ^^^^^^^^^^ use of undeclared crate or module `ptyprocess`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0432`.
Couldn't compile the test.

failures:
    server/src/pty.rs - pty::PtyProcess::spawn (line 48)

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.08s
```

Comment (SuperSandro2000) [reply pkgs/by-name/jo/job-security/package.nix:20]: Oh, there is a rust crate also named like that. That's indeed an upstream error.

Comment (fgaz) [reply pkgs/by-name/jo/job-security/package.nix]: Fixed upstream

**Pull request merged**